src/gsi/gsi_fedOper.F90

```diff
@@ -0,0 +1,185 @@
+    integer(i_kind):: nreal,nchanl,ier,nele
+    logical:: diagsave
+    integer(i_kind):: lu_diag
+    character(128):: diag_file
+    character(80):: string
+
+    if(nobs == 0) then
+
+       if( (mype == 0) .and. init_pass ) then
+          write(string,600) jiter
+600       format('fed_',i2.2)
+          diag_file=trim(dirname) // trim(string)
+          write(6,*) 'write ianldate to ', diag_file
+          open(newunit=lu_diag,file=trim(diag_file),form='unformatted',status='unknown',position='rewind')
+          write(lu_diag) ianldate
+          close(lu_diag)
+       endif
+
+       return
+
+    endif
+
+    read(lunin,iostat=ier) obstype,isis,nreal,nchanl
+    if(ier/=0) call die(myname_,'read(obstype,...), iostat =',ier)
+    nele = nreal+nchanl
+
+    diagsave  = write_diag(jiter) .and. diag_fed
+
+    call setup(self%obsLL(:), self%odiagLL(:), &
+        lunin,mype,bwork,awork(:,iwork),nele,nobs,is,diagsave,init_pass)
+
+  end subroutine setup_
+
+  subroutine intjo1_(self, ibin, rval,sval, qpred,sbias)
+!    use intfedmod, only: intjo => intfed
+    use gsi_bundlemod  , only: gsi_bundle
+    use bias_predictors, only: predictors
+    use m_obsNode , only: obsNode
+    use m_obsLList, only: obsLList_headNode
+    use kinds     , only: i_kind, r_quad
+    implicit none
+    class(fedOper  ),intent(in   ):: self
+    integer(i_kind ),intent(in   ):: ibin
+    type(gsi_bundle),intent(inout):: rval   ! (ibin)
+    type(gsi_bundle),intent(in   ):: sval   ! (ibin)
+    real(r_quad    ),target,dimension(:),intent(inout):: qpred  ! (ibin)
+    type(predictors),target,             intent(in   ):: sbias
+
+    !----------------------------------------
+    character(len=*),parameter:: myname_=myname//"::intjo1_"
+    class(obsNode),pointer:: headNode
+
+!    headNode => obsLList_headNode(self%obsLL(ibin))
+!    call intjo(headNode, rval,sval)
+!    headNode => null()
+
+  end subroutine intjo1_
+
+  subroutine stpjo1_(self, ibin, dval,xval,pbcjo,sges,nstep,dbias,xbias)
+!    use stpfedmod, only: stpjo => stpfed
+    use gsi_bundlemod, only: gsi_bundle
+    use bias_predictors, only: predictors
+    use m_obsNode , only: obsNode
+    use m_obsLList, only: obsLList_headNode
+    use kinds, only: r_quad,r_kind,i_kind
+    implicit none
+    class(fedOper  ),intent(in):: self
+    integer(i_kind ),intent(in):: ibin
+    type(gsi_bundle),intent(in):: dval
+    type(gsi_bundle),intent(in):: xval
+    real(r_quad    ),dimension(:),intent(inout):: pbcjo ! (1:4)
+    real(r_kind    ),dimension(:),intent(in   ):: sges
+    integer(i_kind),intent(in):: nstep
+
+    type(predictors),target, intent(in):: dbias
+    type(predictors),target, intent(in):: xbias
+
+    !----------------------------------------
+    character(len=*),parameter:: myname_=myname//"::stpjo1_"
+    class(obsNode),pointer:: headNode
+
+!    headNode => obsLList_headNode(self%obsLL(ibin))
+!    call stpjo(headNode,dval,xval,pbcjo(:),sges,nstep)
+!    headNode => null()
```

Comment:
Line 180-182: Similar as above, suggest removing them if not needed.

Reply:
Agreed.